tsdb/store.go

```diff
@@ -813,13 +813,20 @@ func (s *Store) DeleteShard(shardID uint64) error {
 		return nil
 	})
 
+	if err != nil {
+		s.Logger.Error("error walking shards during deletion", zap.Error(err))
```

Comment:
It would be good to reference the operation underway.  The error messages from `sh.Index()` are descriptive, but let's tell the user that we are deleting a shard, or in `DeleteShard` or something for more context.